File: apk-out/sources/com/nick/myrecoverytracker/LocationWorker.java
package com.nick.myrecoverytracker;

import android.content.Context;
import android.location.Location;
import android.location.LocationManager;
import android.util.Log;
import androidx.constraintlayout.widget.ConstraintLayout;
import androidx.core.content.ContextCompat;
import androidx.work.ListenableWorker;
import androidx.work.Worker;
import androidx.work.WorkerParameters;
import java.io.File;
import java.io.FileWriter;
import java.io.IOException;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.List;
import java.util.Locale;
import kotlin.Metadata;
import kotlin.jvm.internal.Intrinsics;

/* compiled from: LocationWorker.kt */
@Metadata(d1 = {"\u0000.\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0003\n\u0002\u0018\u0002\n\u0000\n\u0002\u0010\u0002\n\u0002\b\u0002\n\u0002\u0010\u000e\n\u0002\b\u0002\u0018\u00002\u00020\u0001B\u0017\u0012\u0006\u0010\u0002\u001a\u00020\u0003\u0012\u0006\u0010\u0004\u001a\u00020\u0005¢\u0006\u0004\b\u0006\u0010\u0007J\b\u0010\b\u001a\u00020\tH\u0016J\b\u0010\n\u001a\u00020\u000bH\u0003J\u0018\u0010\f\u001a\u00020\u000b2\u0006\u0010\r\u001a\u00020\u000e2\u0006\u0010\u000f\u001a\u00020\u000eH\u0002¨\u0006\u0010"}, d2 = {"Lcom/nick/myrecoverytracker/LocationWorker;", "Landroidx/work/Worker;", "appContext", "Landroid/content/Context;", "workerParams", "Landroidx/work/WorkerParameters;", "<init>", "(Landroid/content/Context;Landroidx/work/WorkerParameters;)V", "doWork", "Landroidx/work/ListenableWorker$Result;", "logLocation", "", "appendToFile", "filename", "", "line", "app_debug"}, k = 1, mv = {2, 0, 0}, xi = ConstraintLayout.LayoutParams.Table.LAYOUT_CONSTRAINT_VERTICAL_CHAINSTYLE)
/* loaded from: classes3.dex */
public final class LocationWorker extends Worker {
    /* JADX WARN: 'super' call moved to the top of the method (can break code semantics) */
    public LocationWorker(Context appContext, WorkerParameters workerParams) {
        super(appContext, workerParams);
        Intrinsics.checkNotNullParameter(appContext, "appContext");
        Intrinsics.checkNotNullParameter(workerParams, "workerParams");
    }

    @Override // androidx.work.Worker
    public ListenableWorker.Result doWork() throws IOException {
        logLocation();
        ListenableWorker.Result resultSuccess = ListenableWorker.Result.success();
        Intrinsics.checkNotNullExpressionValue(resultSuccess, "success(...)");
        return resultSuccess;
    }

    private final void logLocation() throws IOException {
        Context context = getApplicationContext();
        Intrinsics.checkNotNullExpressionValue(context, "getApplicationContext(...)");
        Object systemService = context.getSystemService("location");
        Intrinsics.checkNotNull(systemService, "null cannot be cast to non-null type android.location.LocationManager");
        LocationManager locationManager = (LocationManager) systemService;
        boolean hasPermission = ContextCompat.checkSelfPermission(context, "android.permission.ACCESS_FINE_LOCATION") == 0;
        if (!hasPermission) {
            Log.w("LocationWorker", "🚫 Missing ACCESS_FINE_LOCATION permission");
            return;
        }
        List providers = locationManager.getProviders(true);
        Intrinsics.checkNotNullExpressionValue(providers, "getProviders(...)");
        Location bestLocation = null;
        for (String provider : providers) {
            Location location = locationManager.getLastKnownLocation(provider);
            if (location != null && (bestLocation == null || location.getAccuracy() < bestLocation.getAccuracy())) {
                bestLocation = location;
            }
        }
        if (bestLocation == null) {
            Log.w("LocationWorker", "⚠️ No location data available");
            return;
        }
        String timestamp = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss", Locale.US).format(new Date());
        String logLine = timestamp + "," + bestLocation.getLatitude() + "," + bestLocation.getLongitude() + "," + bestLocation.getAccuracy();
        Log.i("LocationWorker", "📍 Location logged: " + logLine);
        appendToFile("location_log.csv", logLine);
    }

    private final void appendToFile(String filename, String line) throws IOException {
        try {
            File file = new File(getApplicationContext().getFilesDir(), filename);
            FileWriter writer = new FileWriter(file, true);
            Appendable appendableAppend = writer.append((CharSequence) line);
            Intrinsics.checkNotNullExpressionValue(appendableAppend, "append(...)");
            Intrinsics.checkNotNullExpressionValue(appendableAppend.append('\n'), "append(...)");
            writer.flush();
            writer.close();
        } catch (Exception e) {
            Log.e("LocationWorker", "❌ Failed to write location log", e);
        }
    }
}

File: app/src/main/java/com/nick/myrecoverytracker/LocationWorker.kt
package com.nick.myrecoverytracker

import android.Manifest
import android.annotation.SuppressLint
import android.content.Context
import android.location.Location
import android.location.LocationManager
import android.util.Log
import androidx.core.content.ContextCompat
import androidx.core.content.PermissionChecker
import androidx.work.Worker
import androidx.work.WorkerParameters
import java.io.File
import java.io.FileWriter
import java.text.SimpleDateFormat
import java.util.*

class LocationWorker(appContext: Context, workerParams: WorkerParameters) :
    Worker(appContext, workerParams) {

    override fun doWork(): Result {
        logLocation()
        return Result.success()
    }

    @SuppressLint("MissingPermission")
    private fun logLocation() {
        val context = applicationContext
        val locationManager = context.getSystemService(Context.LOCATION_SERVICE) as LocationManager

        val hasPermission = ContextCompat.checkSelfPermission(context, Manifest.permission.ACCESS_FINE_LOCATION) ==
                PermissionChecker.PERMISSION_GRANTED

        if (!hasPermission) {
            Log.w("LocationWorker", "🚫 Missing ACCESS_FINE_LOCATION permission")
            return
        }

        val providers = locationManager.getProviders(true)
        var bestLocation: Location? = null

        for (provider in providers) {
            val location = locationManager.getLastKnownLocation(provider)
            if (location != null && (bestLocation == null || location.accuracy < bestLocation.accuracy)) {
                bestLocation = location
            }
        }

        if (bestLocation != null) {
            val timestamp = SimpleDateFormat("yyyy-MM-dd HH:mm:ss", Locale.US).format(Date())
            val logLine = "$timestamp,${bestLocation.latitude},${bestLocation.longitude},${bestLocation.accuracy}"
            Log.i("LocationWorker", "📍 Location logged: $logLine")
            appendToFile("location_log.csv", logLine)
        } else {
            Log.w("LocationWorker", "⚠️ No location data available")
        }
    }

    private fun appendToFile(filename: String, line: String) {
        try {
            val file = File(applicationContext.filesDir, filename)
            val writer = FileWriter(file, true)
            writer.appendLine(line)
            writer.flush()
            writer.close()
        } catch (e: Exception) {
            Log.e("LocationWorker", "❌ Failed to write location log", e)
        }
    }
}